packages/sync/src/links/OfflineQueueLink.ts

```diff
@@ -153,4 +171,21 @@ export class OfflineQueueLink extends ApolloLink {
       });
     }
   }
+
+  private updateIds(
+    operationEntry: OperationQueueEntry,
+    result: FetchResult
+  ): void {
+    const operation = operationEntry.operation;
+    const optimisticResponse = operationEntry.optimisticResponse;
+    if (optimisticResponse && optimisticResponse[operation.operationName].__optimisticId) {
```

Comment:
[Idea] Instead of flag we can simply have `__optimisticId` or `__clientId` field. This will be more beneficial because at any point we will be able to get both server and client id.